hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/security/UserGroupInformation.java

```diff
@@ -1968,6 +1976,7 @@ private static UserGroupInformation doSubjectLogin(
       if (subject == null) {
         params.put(LoginParam.PRINCIPAL, ugi.getUserName());
         ugi.setLogin(login);
+        ugi.setLastLogin(Time.now());
```

Comment:
the `user::lastLogin` was the last attempt to login. I'm not sure if we should put this before the `try` clause (aka before L1969)?

CC: @steveloughran

Reply:
@liuml07 We should set this only after the login is successful right ? If we add it before the try, then we might update it even in case of unsuccessful login in which case a retry should be permitted immediately. Please let me know your thoughts